File: backend/src/modules/app-config/app-config.model.ts
import { Schema, model, Document } from 'mongoose';

export interface IAppConfig extends Document {
    maintenanceMode: boolean;
    maintenanceMessage?: string;
    minAppVersion: string;
    latestAppVersion: string;
    forceUpdate: boolean;
    socialLinks: {
        instagram?: string;
        telegram?: string;
        whatsapp?: string;
        youtube?: string;
    };
    supportEmail: string;
    privacyPolicyUrl: string;
    termsConditionsUrl: string;
    primaryColor: string;
    featureFlags: Record<string, boolean>;
    createdAt: Date;
    updatedAt: Date;
}

const appConfigSchema = new Schema<IAppConfig>(
    {
        maintenanceMode: { type: Boolean, default: false },
        maintenanceMessage: { type: String, default: 'We are currently under maintenance. Please check back later.' },
        minAppVersion: { type: String, default: '1.0.0' },
        latestAppVersion: { type: String, default: '1.0.0' },
        forceUpdate: { type: Boolean, default: false },
        socialLinks: {
            instagram: { type: String },
            telegram: { type: String },
            whatsapp: { type: String },
            youtube: { type: String },
        },
        supportEmail: { type: String, required: true, default: 'support@examease.com' },
        privacyPolicyUrl: { type: String, default: '' },
        termsConditionsUrl: { type: String, default: '' },
        primaryColor: { type: String, default: '#1E88E5' },
        featureFlags: { type: Schema.Types.Mixed, default: {} },
    },
    {
        timestamps: true,
    }
);

export const AppConfig = model<IAppConfig>('AppConfig', appConfigSchema);

File: mobile_app/lib/core/models/app_config_model.dart
class AppConfigModel {
  final bool maintenanceMode;
  final String maintenanceMessage;
  final String minAppVersion;
  final String latestAppVersion;
  final bool forceUpdate;
  final String supportEmail;
  final String privacyPolicyUrl;
  final String termsConditionsUrl;
  final String primaryColor;

  AppConfigModel({
    required this.maintenanceMode,
    required this.maintenanceMessage,
    required this.minAppVersion,
    required this.latestAppVersion,
    required this.forceUpdate,
    required this.supportEmail,
    required this.privacyPolicyUrl,
    required this.termsConditionsUrl,
    required this.primaryColor,
  });

  factory AppConfigModel.fromJson(Map<String, dynamic> json) {
    return AppConfigModel(
      maintenanceMode: json['maintenanceMode'] as bool? ?? false,
      maintenanceMessage:
          json['maintenanceMessage'] as String? ?? 'Under maintenance',
      minAppVersion: json['minAppVersion'] as String? ?? '1.0.0',
      latestAppVersion: json['latestAppVersion'] as String? ?? '1.0.0',
      forceUpdate: json['forceUpdate'] as bool? ?? false,
      supportEmail: json['supportEmail'] as String? ?? '',
      privacyPolicyUrl: json['privacyPolicyUrl'] as String? ?? '',
      termsConditionsUrl: json['termsConditionsUrl'] as String? ?? '',
      primaryColor: json['primaryColor'] as String? ?? '#1E88E5',
    );
  }
}
